src/components/icon-button/icon-button.spec.tsx

```diff
@@ -12,17 +12,67 @@ import StyledIconButton from "./icon-button.style";
 import Icon from "../icon";
 import StyledIcon from "../icon/icon.style";
 import { TooltipProvider } from "../../__internal__/tooltip-provider";
+import Logger from "../../__internal__/utils/logger";
+
+jest.mock("../../__internal__/utils/logger");
 
 describe("IconButton component", () => {
   let wrapper: ReactWrapper;
   let onDismiss: jest.Mock;
+  let loggerSpy: jest.SpyInstance<void, [message: string]> | jest.Mock;
+
+  describe("deprecation warning", () => {
+    it("should display deprecation warning when the onAction is passed to the component is used once", () => {
+      loggerSpy = jest.spyOn(Logger, "deprecate");
```

Comment:
```suggestion
  describe("deprecation warning", () => {
    it("should display deprecation warning when the onAction is passed to the component is used once", () => {
      const loggerSpy = jest.spyOn(Logger, "deprecate");
```
For readability I would move the scope of `loggerSpy` to inside the test since it is not used elsewhere